src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -93,25 +77,25 @@ class DataModelInstanceRelation(config: RelationConfig, modelExternalId: String)
   def getInstanceFilter(sparkFilter: Filter): Option[DataModelInstanceFilter] =
     sparkFilter match {
       case EqualTo(left, right) => {
-        Some(DMIEqualsFilter(Seq(modelExternalId, left), parseValue(right)))
+        Some(DMIEqualsFilter(Seq(modelExternalId, left), parsePropertyValue(right)))
       }
       case In(attribute, values) =>
-        if (propertyTypes(attribute)._1.endsWith("[]")) {
+        if (modelInfo(attribute).`type`.endsWith("[]")) {
           None
         } else {
           val setValues = values.filter(_ != null)
-          Some(DMIInFilter(Seq(modelExternalId, attribute), setValues.map(parseValue)))
+          Some(DMIInFilter(Seq(modelExternalId, attribute), setValues.map(parsePropertyValue)))
         }
       case StringStartsWith(attribute, value) =>
-        Some(DMIPrefixFilter(Seq(modelExternalId, attribute), parseValue(value)))
+        Some(DMIPrefixFilter(Seq(modelExternalId, attribute), parsePropertyValue(value)))
       case GreaterThanOrEqual(attribute, value) =>
-        Some(DMIRangeFilter(Seq(modelExternalId, attribute), gte = Some(parseValue(value))))
+        Some(DMIRangeFilter(Seq(modelExternalId, attribute), gte = Some(parsePropertyValue(value))))
       case GreaterThan(attribute, value) =>
-        Some(DMIRangeFilter(Seq(modelExternalId, attribute), gt = Some(parseValue(value))))
+        Some(DMIRangeFilter(Seq(modelExternalId, attribute), gt = Some(parsePropertyValue(value))))
       case LessThanOrEqual(attribute, value) =>
-        Some(DMIRangeFilter(Seq(modelExternalId, attribute), lte = Some(parseValue(value))))
+        Some(DMIRangeFilter(Seq(modelExternalId, attribute), lte = Some(parsePropertyValue(value))))
       case LessThan(attribute, value) =>
-        Some(DMIRangeFilter(Seq(modelExternalId, attribute), lt = Some(parseValue(value))))
+        Some(DMIRangeFilter(Seq(modelExternalId, attribute), lt = Some(parsePropertyValue(value))))
       case And(f1, f2) =>
         (getInstanceFilter(f1) ++ getInstanceFilter(f2)).reduceLeftOption((sf1, sf2) =>
           DMIAndFilter(Seq(sf1, sf2)))
```

Comment:
I have a doubt about the case Or:
```scala
case Or(f1, f2) =>
    (getInstanceFilter(f1), getInstanceFilter(f2)) match {
      case (Some(sf1), Some(sf2)) =>
        Some(DMIOrFilter(Seq(sf1, sf2)))
      case _ =>
        None
    }
```
Don't we want  to have 2 case below as well ? 
```scala
case (Some(sf1), None) =>
            Some(DMIOrFilter(Seq(sf1)))
case (None, Some(sf2)) =>
            Some(DMIOrFilter(Seq(sf2)))
```

Reply:
I had a reason for this, let me try recalling...

Reply:
So the reason we get `None` here, is we can't match the filter any case here. So we should not push down by only one condition we can match to something. It may be a wider filter when we `or` and we pushdown more than we should. So we let spark filter the whole thing in that case, not to miss requested data.